src/Validators/OptionalIfCheck.cs

```diff
@@ -0,0 +1,63 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using form_builder.Models;
+using form_builder.Models.Elements;
+
+namespace form_builder.Validators.IntegrityChecks.Form
+{
+    public class OptionalIfCheck : IFormSchemaIntegrityCheck
+    {
+        public IntegrityCheckResult Validate(FormSchema schema)
+        {
+            IntegrityCheckResult result = new();
+            List<string> questionIds = new();
+
+            List<IElement> elementsWithOptionalIfInJSON = schema.Pages
+                .Where(page => page.Elements is not null)
+                .SelectMany(page => page.Elements)
+                .Where(element => !string.IsNullOrEmpty(element.Properties.OptionalIf.QuestionId) || !string.IsNullOrEmpty(element.Properties.OptionalIf.ComparisonValue) || !element.Properties.OptionalIf.ConditionType.Equals(Enum.ECondition.Undefined))
+                .ToList();
```

Comment:
If you made it an IEnumrable, you wouldn't need to call it ToList().... but you would need to change the IF statement below to "!.Any()".  Could save some unnecessary work.